dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionHelper.java

```diff
@@ -101,6 +101,7 @@ public void evictAndAddEntries(Collection collection) {
             entry.setVerified(!versionDAO.findEntryVersionsWithVerifiedPlatforms(entry.getId()).isEmpty());
             List<CategorySummary> summaries = entryDAO.findCategorySummariesByEntryId(entry.getId());
             entry.setCategorySummaries(summaries);
+            entry.setTopic(entryDAO.getGenericEntryById(entry.getId()).getTopic());
```

Comment:
Ooof, was like this already but there's probably a cleaner way of doing this. But it matches the pattern. 
In the meantime, please add a test to lock this down. 

One place to start looking could be https://github.com/dockstore/dockstore/blob/881b61a05a4e3edb7b545503a17e4d4a65b32756/dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java#L560

Reply:
I think the way to do this would be to instead update the JPA queries starting around [here](https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java#L121), but do correct me if I'm wrong.

I think the reason for this pattern is that it's hard to do JPA Data Transfer Object (DTO) projections with "many" values in a single query; that's the many fields are being set with separate queries. Not sure about `entryType` though and why it's not set in the query; it does look the the constructor for `CollectionEntry` is getting pretty large.

Reply:
Barring a change to the queries themselves, is the theory that since [on this line] we're now retrieving the Entry itself, we can probably get the remaining information out of the Entry, thus clarifying the code, while inducing no more db queries than happened before?

Reply:
Does this method exist because we used to eagerly fetch versions and/or some calculated fields iterated through all the versions? I assume that's still applicable? Just checking because at some point we made versions lazy and I wonder if we still need this (we probably do).

Reply:
Alternatively, we could create a query like `entryDAO.getTopic(entry.getId())` so we're not retrieving the entire entry, which it seems like we're trying to avoid based on how we set descriptor types below. The query might need to use `CASE` and return the appropriate column based on the value of `topicSelection`

Not sure if it's worth it right now though